clang/test/Driver/flang/classic-flang.f

```diff
@@ -5,22 +5,22 @@
 
 ! RUN: %clang --driver-mode=flang -target x86_64-unknown-linux-gnu -c %s -### 2>&1 \
 ! RUN:   | FileCheck %s
-! CHECK: "flang1"
+! CHECK: "{{.*}}flang1"
 ! CHECK-NOT: "-preprocess"
 ! CHECK-SAME: "-nofreeform"
-! CHECK-NEXT: "flang2"
+! CHECK-NEXT: "{{.*}}flang2"
 ! CHECK-NEXT: {{clang.* "-cc1"}}
 
 ! Check that the driver invokes flang1 correctly when preprocessing is
 ! explicitly requested.
 
 ! RUN: %clang --driver-mode=flang -target x86_64-unknown-linux-gnu -E %s -### 2>&1 \
 ! RUN:   | FileCheck --check-prefix=CHECK-PREPROCESS %s
-! CHECK-PREPROCESS: "flang1"
+! CHECK-PREPROCESS: "{{.*}}flang1"
 ! CHECK-PREPROCESS-SAME: "-preprocess"
 ! CHECK-PREPROCESS-SAME: "-es"
 ! CHECK-PREPROCESS-SAME: "-pp"
-! CHECK-PREPROCESS-NOT: "flang1"
-! CHECK-PREPROCESS-NOT: "flang2"
+! CHECK-PREPROCESS-NOT: "{{^.*}}flang1"
+! CHECK-PREPROCESS-NOT: "{{^.*}}flang2"
```

Comment:
Question: Why is the `^` needed in the regexps here? I removed the caret and the tests still work.

Reply:
Indeed, the caret should not be here. I added it in the beginning, but when I got how it works I removed it from all expressions. I missed this one usage. 
I don't want to set up a new PR now, just for this one change. Can we agree that whoever changes the tests in the nearest future removes this caret? :)

Reply:
sadly, this is a recipe for leaving it like that forever.

Reply:
@pawosm-arm It would have been nice to not accept pull requests that have open, unresolved discussions.

Reply:
But then maybe I shouldn't have approved it preemptively.

Reply:
It is trivial to fix the errors, so I just set up two more PRs:
https://github.com/flang-compiler/classic-flang-llvm-project/pull/29
https://github.com/flang-compiler/classic-flang-llvm-project/pull/30
Apologies for not noticing this earlier and thus - polluting the commit history.
@bryanpkc , in case you cherry-pick this to `release_12x`, please feel free to squash the extra commit with the original one.